Formula/openssl110.rb

```diff
@@ -0,0 +1,184 @@
+    no-ssl3
+    no-ssl3-method
+    no-zlib
+    shared
+    enable-cms
+  ]
+  end
+
+  def install
+    if build.universal?
+      ENV.permit_arch_flags
+      archs = Hardware::CPU.universal_archs
+    elsif MacOS.prefer_64_bit?
+      archs = [Hardware::CPU.arch_64_bit]
+    else
+      archs = [Hardware::CPU.arch_32_bit]
+    end
+
+    dirs = []
+
+    archs.each do |arch|
+      if build.universal?
+        dir = "build-#{arch}"
+        dirs << dir
+        mkdir dir
+        mkdir "#{dir}/engines"
+      end
+
+      ENV.deparallelize
+      system "perl", "./Configure", *(configure_args + arch_args[arch])
+      system "make", "clean" if build.universal?
+      system "make", "depend"
+      system "make"
+      system "make", "test" if build.with?("test")
+
+      if build.universal?
+        cp "include/openssl/opensslconf.h", dir
+        cp Dir["*.?.?.dylib", "*.a", "apps/openssl"], dir
+        cp Dir["engines/**/*.dylib"], "#{dir}/engines"
+      end
+    end
+
+    system "make", "install", "MANDIR=#{man}", "MANSUFFIX=ssl"
+
+    if build.universal?
+      %w[libcrypto libssl].each do |libname|
+        system "lipo", "-create", "#{dirs.first}/#{libname}.1.1.dylib",
+                                  "#{dirs.last}/#{libname}.1.1.dylib",
+                       "-output", "#{lib}/#{libname}.1.1.dylib"
+        system "lipo", "-create", "#{dirs.first}/#{libname}.a",
+                                  "#{dirs.last}/#{libname}.a",
+                       "-output", "#{lib}/#{libname}.a"
+      end
+
+      Dir.glob("#{dirs.first}/engines/*.dylib") do |engine|
+        libname = File.basename(engine)
+        system "lipo", "-create", "#{dirs.first}/engines/#{libname}",
+                                  "#{dirs.last}/engines/#{libname}",
+                       "-output", "#{lib}/engines/#{libname}"
+      end
+
+      system "lipo", "-create", "#{dirs.first}/openssl",
+                                "#{dirs.last}/openssl",
+                     "-output", "#{bin}/openssl"
+
+      confs = archs.map do |arch|
+        <<-EOS.undent
+          #ifdef __#{arch}__
+          #{(buildpath/"build-#{arch}/opensslconf.h").read}
+          #endif
+          EOS
+      end
+      (include/"openssl/opensslconf.h").atomic_write confs.join("\n")
+    end
+  end
+
+  def openssldir
+    etc/"openssl110"
+  end
+
+  def post_install
+    # OpenSSL "remembers" the Perl it was built against. Where that is
+    # Homebrew's Perl it remembers the Cellar path, which breaks every version.
+    # Fix this manually to use the opt path which remains consistent.
+    perl_scripts = %W[
+      #{openssldir}/misc/CA.pl
+      #{openssldir}/misc/tsget
+      #{bin}/c_rehash
+    ]
+    perl_scripts.each do |f|
+      next unless File.exist?(f)
+      if File.read(f).include?(Formula["perl"].bin)
+        inreplace f, Formula["perl"].bin, Formula["perl"].opt_bin
+      end
+    end
+
+    keychains = %w[
+      /Library/Keychains/System.keychain
```

Comment:
Good. 

> Never trust an exposed connection facing the internet, sadly.
I feel you, but note that (in theory) off-the-shelf Server.app is supposed to be secure.

I was in the middle of editing my "demo comment" when you cleaned up, so you can take it or leave it as you like, if you think its helpful.

This line is not such a good idea.

As in:

Example:
1. Create a self-signed CA, and then create and sign a SSL cert with that CA. Launch a https server that uses these (Server.app is a quick way to do this, especially as we're using SecureTransport to contrast with OpenSSL). Once configure correctly, when you browse to your server, you should get a big invalid certificate modal dialog.

2. User the system `curl` to hit it over HTTPS.
You should also get stopped here because of certificate problems.

3. Reinstall openssl.rb, so the keychains get dumped again.  Then, using Homebrew curl (or wget, etc., anything linked to brew's OpenSSL), provided everything is where it should be, you should now no longer get an error. 
Not a good thing. 😨